docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md

```diff
@@ -0,0 +1,384 @@
+) -> Iterable[DltResource]:
+```
+
+Args:
+
+`connection_url`: MongoDB connection URL.
+
+`database`: Database name (defaults if unspecified).
+
+`collection_names`: Names of desired collections; loads all if not specified.
+
+`incremental`: Option for incremental data loading.
+
+`write_disposition`: Writing mode: "replace", "append", or "merge".
+
+Returns:
+
+`Iterable[DltResource]`: A list of DLT resources for each specified collection.
+
+### Resource `collection.name`
+
+This code segment is part of the source "mongodb", processes a list of MongoDB collections
+(collection_list). For each collection, a DLT resource is setup using "dlt.resource", with
+"collection_documents" as the data-yielding function.
+
+```python
+for collection in collection_list:
+    yield dlt.resource(  # type: ignore
+        collection_documents,
+        name=collection.name,
+        primary_key="_id",
+        write_disposition=write_disposition,
+        spec=MongoDbCollectionConfiguration,
+    )(client, collection, incremental=incremental)
+```
+
+`name`: The current collection's name.
+
+`primary_key`: '_id' is the default MongoDB primary key.
+
+`write_disposition`: Writing mode: "replace", "append", or "merge".
+
+The `spec` parameter deploys 'MongoDbCollectionConfiguration' for MongoDB setup. Each resource is
+invoked with the MongoDB client, current collection, and incremental parameter, preparing a DLT
+resource for each item in collection_list.
+
+### Source `mongo_collection`
+
+This function fetches data from a MongoDB collection, returning resources for each collection to be loaded.
+
+```python
+def mongodb_collection(
+    connection_url: str = dlt.secrets.value,
+    database: Optional[str] = dlt.config.value,
+    collection: str = dlt.config.value,
+    incremental: Optional[dlt.sources.incremental] = None,  # type: ignore[type-arg]
+    write_disposition: Optional[str] = dlt.config.value,
+) -> Any:
+```
+
+Args:
+
+The arguments `connection_url`, `database`, `incremental` and `write_disposition` are the same as
+defined for [Source `mongodb`](#source-mongodb).
+
+`collection`: Name of the collection to load.
+
+Returns:
+
+Iterable [DltResource]: A list of DLT resources ("collection_documents") for each collection to be
+loaded.
+
+### Resource `collection_obj.name`
+
+This function, part of the "mongo_collection" source, loads documents from a collection. Each
+resource is invoked with the MongoDB client, collection object, and incremental parameter, preparing
+a DLT resource for each item in the collection.
+
+```python
+return dlt.resource(  # type: ignore
+    collection_documents,
+    name=collection_obj.name,
+    primary_key="_id",
+    write_disposition=write_disposition,
+)(client, collection_obj, incremental=incremental)
+```
+
+`name`: Collection object's name.
+
+Parameters `primary_key` and `write_disposition` are as mentioned in
+[resource `collection_documents`.](#resource-collectiondocuments)
+
+### Create your own pipeline
```

Comment:
i would recommend take the existed README as an example, there is good style.
https://github.com/dlt-hub/verified-sources/blob/master/sources/mongodb/README.md#how-to-use

Reply:
deleted